src/algorithms/path.c

```diff
@@ -0,0 +1,63 @@
+/*
+* Copyright 2018-2019 Redis Labs Ltd. and Contributors
+*
+* This file is available under the Apache License, Version 2.0,
+* modified with the Commons Clause restriction.
+*/
+
+#include "./path.h"
+#include "../util/arr.h"
+
+Path Path_new(size_t len) {
+    Path p = array_new(Edge*, len);
+    return p;
+}
+
+Path Path_append(Path p, Edge *e) {
+    p = array_append(p, e);
+    return p;
+}
+
+Edge *Path_pop(Path p) {
+    size_t pathLen = array_len(p);
+    if(pathLen == 0) return NULL;
+    
+    Edge *e = p[pathLen-1];
+    array_pop(p);
```

Comment:
These two lines are equivalent to:
`Edge *e = array_pop(p);`
Or can even `return array_pop(p)`